execution/scan/matrix_selector.go

```diff
@@ -155,13 +156,14 @@ func (o *matrixSelector) Next(ctx context.Context) ([]model.StepVector, error) {
 			// Also, allow operator to exist independently without being nested
 			// under parser.Call by implementing new data model.
 			// https://github.com/thanos-community/promql-engine/issues/39
-			result := o.call(function.FunctionArgs{
-				Labels:      series.labels,
-				Samples:     rangeSamples,
-				StepTime:    seriesTs,
-				SelectRange: o.selectRange,
-				Offset:      o.offset,
-			})
+			fa := function.NewFunctionsArgs()
+			fa.Labels = series.labels
```

Comment:
Does it make sense to have a constructor with all parameters so we don't accidentally omit something?